week10/src/pages/Detail.jsx

```diff
@@ -0,0 +1,392 @@
+import { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { GetPostDetailApi, EditPostApi, DeletePostApi } from "../api/post";
+import {
+    postHeart,
+    deleteHeart,
+    getHeartNum,
+    checkMyClick,
+} from "../api/heart";
+import { deleteComments, getComments, postComments } from "../api/comment";
+import { JWTtest } from "../api/user";
+import styled from "styled-components";
+
+import heart from "../img/heart.png";
+
+const Detail = () => {
+    const { postId } = useParams();
+    const [details, setDetails] = useState(null);
+    const [isEdit, setIsEdit] = useState(false);
+    const [post, setPost] = useState({ title: "", content: "" });
+    const [heartNum, setHeartNum] = useState(0);
+    const [myHeart, setMyheart] = useState(false);
+    const [myComments, setMyComments] = useState("");
+    const [commentsList, setCommentsList] = useState(null);
+    const [myAccount, setMyAccount] = useState("");
+    const nav = useNavigate();
+
+    useEffect(() => {
+        async function getDetails() {
+            const res = await GetPostDetailApi(postId);
+            setDetails(res);
+            const res2 = await JWTtest();
+
+            setMyAccount(res2.nickname);
+        }
```

Comment:
```suggestion
    useEffect(() => {
        GetPostDetailApi(postId)
            .then(res => setDetails(res))
            .catch(err => console.log(err));
        JWTtest()
            .then(res => setMyAccount(res.nickname));
```
이번 과제에서는 외부 파일에서 이미 API 통신을 하는 함수를 async await으로 정의했기 때문에 함수를 단순히 호출하여 사용하는 파일에서 이렇게 또 정의할 필요가 없답니다!
위 예시처럼 then-catch문을 사용해서 간단하게 작성하거나, 함수 호출 앞뒤로 추가 작업이 많이 필요하다면 useEffect 외부에서 async 함수를 정의하고 내부에서는 호출문만 작성하여 간결하게 유지해주세요.